src/app/public/modules/popover/popover.directive.ts

```diff
@@ -132,80 +133,98 @@ export class SkyPopoverDirective implements OnChanges, OnDestroy {
     }
   }
 
-  private isPopoverOpen(): boolean {
-    return (this.skyPopover && this.skyPopover.isOpen);
-  }
-
   private addEventListeners(): void {
     const element = this.elementRef.nativeElement;
 
     this.skyPopoverMessageStream
-      .takeUntil(this.idled)
+      .takeUntil(this.ngUnsubscribe)
       .subscribe(message => {
         this.handleIncomingMessages(message);
       });
 
+    Observable
+      .fromEvent(element, 'keydown')
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((event: KeyboardEvent) => {
+        if (!this.skyPopover.isActive) {
+          return;
+        }
+
+        const key = event.key.toLowerCase();
+
+        /* tslint:disable-next-line:switch-default */
+        switch (key) {
+          case 'arrowdown':
+          case 'arrowleft':
+          case 'arrowright':
+          case 'arrowup':
+          case 'down':
+          case 'left':
+          case 'right':
+          case 'up':
+            this.sendMessage(SkyPopoverMessageType.Focus);
```

Comment:
In the current popover, the arrow keys will NOT focus on the popover. Is this change deliberate?

Reply:
Yes. Today, when the user presses the "tab" key, the popover content receives focus (since it exists in the DOM directly after the button). However, according to W3C the tab key should not bring focus to a "modal" or tooltip component (instead, the modal should immediately receive focus when it is opened; however, this is not an expected behavior of popovers today). I added arrow key focusability so that the user could use the keyboard to navigate to the popover (which is now in an overlay). I checked our issue history and it turns out that popovers never received an a11y audit from Matt Gregg. It might be worth reevaluating the popover's interactions to meet W3C guidelines.

Reply:
Sounds good. Thanks for looking into this. I can confirm the arrow keys work as you described. 👍